analyses/survival-analysis/survival-analysis_histology.Rmd

```diff
@@ -0,0 +1,202 @@
+indep_rna <- readr::read_tsv(indep_file)
+
+# remove cell lines by selecting only solid tissue
+histologies <- readr::read_tsv(metadata_file, guess_max = 10000) %>%
+ dplyr::filter(composition=="Solid Tissue" & tumor_descriptor == "Initial CNS Tumor") %>%
+ select(sample_id, broad_histology, short_histology, cancer_group, molecular_subtype, OS_status, OS_days, PFS_days) %>%
+ distinct()
+
+# read in tp53 scores
+tp53_scores <- readr::read_tsv(tp53_file) %>%
+ select(Kids_First_Biospecimen_ID_DNA, Kids_First_Biospecimen_ID_RNA, sample_id, tp53_score)
+
+# read in telomerase scores
+tel_scores <- readr::read_tsv(tel_stranded_file) %>%
+ mutate(Kids_First_Biospecimen_ID_RNA = SampleID,
+        telomerase_score = NormEXTENDScores) %>%
+ select(Kids_First_Biospecimen_ID_RNA, telomerase_score)
+
+# remove samples without RNA-Seq since both classifiers' results use RNA-Seq
+# independent use primary plus rna-seq samples
+meta_indep <- tp53_scores %>%
+  left_join(tel_scores) %>%
+  left_join(histologies) %>%
+  dplyr::filter(Kids_First_Biospecimen_ID_RNA %in% indep_rna$Kids_First_Biospecimen_ID) %>%
+  # recode the categories
+  dplyr::mutate(OS_status_fixed = case_when(
+    OS_status == "LIVING" ~ 0,
+    OS_status == "DECEASED" ~1
+ ))
+
+```
+ 
+## Cox Regression 
+
+### TP53 scores & Telomerase score - results
+```{r Run Cox regression model for tp53 and telomerase}
+# run the model with different covariates
+for(ind_var in c("tp53_score", "telomerase_score")){
+  model <- paste0("survival::Surv(OS_days, OS_status_fixed) ~ ", ind_var)
+  
+  fit <- survival::coxph(
+        formula(model),
+        data = meta_indep
+      )
+  # generate output
+  table <- broom::tidy(fit) 
+  
+  # Save the table data in a TSV
+  readr::write_tsv(table, file.path(results_dir, paste0("cox_reg_results_per_", ind_var, ".tsv")))
+  
+  print(table)
+}
+
+```
+
+## Log Rank analysis 
+
+### Categorical variable cancer group and HGG or not 
+```{r}
+# separate into HGG or non-HGG
+meta_indep <- meta_indep %>% 
+  dplyr::mutate(hgg_group = case_when(
+    short_histology == "HGAT" ~ "HGAT",
+    short_histology != "HGAT" ~ "non-HGAT"
+  ))
+
+# filter to contain only samples in short histology that that have >=10 samples
+meta_indep_sh_filtered <- meta_indep %>% 
+  dplyr::group_by(short_histology) %>% 
+  dplyr::mutate(n_cg = n()) %>% 
+  dplyr::filter(n_cg >=10) 
+
+```
+
+### Generate output for categorical files 
+```{r}
+ind_var_list <- c("hgg_group", "short_histology")
+meta_data_list <- list(meta_indep, meta_indep_sh_filtered)
+
+for(i in 1:length(ind_var_list)){
+  ind_var <- ind_var_list[i]
+  meta_data <- meta_data_list[[i]]
+  
+  # define model
+  model <- paste0("survival::Surv(OS_days, OS_status_fixed) ~ ", ind_var)
+  
+  # run survival analysis
+  fit <- survival::survdiff(formula(model),
+                            data = meta_data)
+  # Obtain p value for Chi-Squared stat
+  fit$p.value <- pchisq(fit$chisq, df = 1, lower = FALSE)
```

Comment:
Seems like this might be a more general solution we can use in the future analyses ([ref](https://stackoverflow.com/questions/36368652/how-to-get-survdiff-returned-p-value/36369255)) ?

```suggestion
  fit$p.value <- pchisq(fit$chisq, df = length(diff$n) - 1, lower = FALSE)
```

Reply:
I modified to `length(fit$n)-1` - should be correct now.